test/behavior/zon.zig

```diff
@@ -0,0 +1,268 @@
+const std = @import("std");
+
+const expect = std.testing.expect;
+const expectEqual = std.testing.expectEqual;
+const expectEqualDeep = std.testing.expectEqualDeep;
+const expectEqualSlices = std.testing.expectEqualSlices;
+const expectEqualStrings = std.testing.expectEqualStrings;
+
+test "void" {
+    try expectEqual({}, @import("zon/void.zon"));
+}
+
+test "bool" {
+    try expectEqual(true, @import("zon/true.zon"));
+    try expectEqual(false, @import("zon/false.zon"));
+}
+
+test "optional" {
+    // Coercion
+    const some: ?u32 = @import("zon/some.zon");
+    const none: ?u32 = @import("zon/none.zon");
+    const @"null": @TypeOf(null) = @import("zon/none.zon");
+    try expectEqual(some, 10);
+    try expectEqual(none, null);
+    try expectEqual(@"null", null);
```

Comment:
The arg order of `expectEqual` matters -- it should be `expected` then `actual`. Same thing in a few other places in this file.